### PR TITLE
Add GPU Deployments section to score.rst

### PR DESCRIPTION
**Pull request by ahosler**

### CODE REVIEW

File: docs/source/user_guide/model_registration/_template/score.rst [hunk 1]
```diff
@@ -34,6 +34,43 @@ Refer Cusotmization section for how to change and verify the model artifacts.
...
+
+.. code-block:: python3
+
+    import torch
+    import torchvision
+    from ads.common.model_metadata import UseCaseType
+    from ads.model.framework.pytorch_model import PyTorchModel
+
+    model = torchvision.models.resnet18(pretrained=True)
+    model.to("cuda:0")
```
Comment (shaksing31) [inline docs/source/user_guide/model_registration/_template/score.rst:51]: Would the example code block would include the randomized algo?

Comment (ahosler) [reply docs/source/user_guide/model_registration/_template/score.rst:51]: The randomised algo is generated within the score.py (during the prepare call). We can add another code snippet for that if you think it adds clarity?

File: docs/source/user_guide/model_registration/_template/score.rst [hunk 2]
```diff
@@ -34,6 +34,43 @@ Refer Cusotmization section for how to change and verify the model artifacts.
 
 The ``score.py`` consists of multiple functions among which the ``load_model`` and ``predict`` are most important.
 
+GPU Deployments
+~~~~~~~~~~~~~~~~
+When deploying your TensorFlow or PyTorch models onto a GPU shape, the ADS generated score.py manages GPU integration for you. It will automatically transfer your data to the GPU (or multiple GPUs) and perform the inference on that GPU. When using ADS 2.8.4 or later, any TensorFlow or PyTorch model artifact can be deployed efficiently on either CPU or GPU regardless of how it was trained.
+
+The Model Deployment Service handles parallelization for you. Whether you have a single or multi GPU deployment, the Model Deployment Service will determine how many replicas of your model can be supported, and the auto-generated score.py will randomly assign those replicas across the GPU(s). The following code example registers a PyTorch Model tuned and deployed on GPUs.
```
Comment (qiuosier) [inline docs/source/user_guide/model_registration/_template/score.rst:41]: how does the replicas are determined? We can probably add links to service docs if there is any.

**Review by qiuosier — DISMISSED** [review]

**Review by mrDzurb — DISMISSED** [review]

File: docs/source/user_guide/model_registration/_template/score.rst (hunk 2 above)
Comment (ahosler) [reply docs/source/user_guide/model_registration/_template/score.rst]: I moved the link up above the code example, and added some color to the replica determination. Thanks!

**Review by qiuosier — APPROVED** [review]

**Review by mrDzurb — APPROVED** [review]

**Pull request merged**